action.yml

```
@@ -7,31 +7,30 @@ runs:
       shell: bash
       run: |
         rm -rf '${{github.workspace}}'
-    - name: Clone
+    - name: Init
       shell: bash
       working-directory: '${{runner.workspace}}'
       run: |
-        git clone 'git@github.com:${{github.repository}}'
-    - name: Checkout for PRs
+        git init '${{github.workspace}}'
+    - name: Add origin
       shell: bash
       working-directory: '${{github.workspace}}'
-      if: ${{ github.event_name == 'pull_request' }}
       run: |
-        git checkout '${{github.event.pull_request.head.sha}}'
-        git config user.name "Actions Runner"
-        git config user.email "noreply@example.com"
-        git merge --no-commit 'origin/${{github.base_ref}}'
-    - name: Checkout for branches
+        git remote add origin 'git@github.com:${{github.repository}}'
+    - name: Checkout branch or PR
       shell: bash
       working-directory: '${{github.workspace}}'
-      if: ${{ github.event_name != 'pull_request' }}
       run: |
-        git checkout '${{github.sha}}'
+        git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin '${{github.sha}}'
+        git checkout --force '${{github.sha}}'
+        git config user.name "Actions Runner"
+        git config user.email "noreply@example.com"
     - name: Update submodules
       shell: bash
       working-directory: '${{github.workspace}}'
       run: |
-        git submodule update --init --recursive
+        git submodule sync --recursive
+        git -c protocol.version=2 submodule update --init --force --depth=1 --recursive
```

```suggestion
        git -c protocol.version=2 submodule update --init --force --depth=1 --recursive
```
This one is potentially dangerous: have you tested it in situations where the submodule is pointing to a commit that is no longer at the tip of any transitLib branch?

Github used to simply not advertise the correct objects in this case leading to failure. If you can confirm that this issue is fully fixed with your new technique, then that would be awesome!

---

I think it will handle that scenario, I'll try testing it in USD

---

It was able to checkout `b3fb3c61e4d0a4bf328dbf116289c405a513d941` of transitlib. I don't think it's a tip of anything, not sure how to check tho

---

Yeah it's not a tip as far as I can see. I tried to artificially create one of those "not our ref" problems now but couldn't.